astropy/cosmology/_src/traits/curvature.py

```diff
@@ -0,0 +1,76 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Critical density.
+
+This is private API. See `~astropy.cosmology.traits` for public API.
+
+"""
+
+__all__ = ["CurvatureComponent"]
+
+__doctest_requires__ = {"CurvatureComponent.Ok0": ["numpy>=2.0"]}
+
+from typing import Any
+
+import numpy as np
+import numpy.typing as npt
+from numpy.typing import ArrayLike
+
+from astropy.cosmology._src.utils import aszarr, deprecated_keywords
+from astropy.units import Quantity
+
+
+class CurvatureComponent:
+    """The object has attributes and methods related to the global curvature."""
+
+    Ok0: float | np.floating[Any]
+    """Omega curvature; the effective curvature density/critical density at z=0."""
+
+    @deprecated_keywords("z", since="7.0")
+    def Ok(self, z: Quantity | ArrayLike) -> npt.NDArray | np.floating[Any]:
```

Comment:
Why is it that `NDArray` does not have a type parameter, but `np.floating` does? I'd expect it to be the other way round.

Reply:
Done.